### PR TITLE
[NEURON][PT][BUILD] - PT1.12 on SDK2.6.0

### PR DESCRIPTION
**Pull request by aws-vrnatham**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Update the PT Neuron Training DLC to the latest Neuron SDK2.6.0 release

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pytorch/training/docker/1.12/py3/sdk2.6.0/Dockerfile.neuron [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+ && curl -o ${HOME_DIR}/oss_compliance.zip https://aws-dlinfra-utilities.s3.amazonaws.com/oss_compliance.zip \
+ && unzip ${HOME_DIR}/oss_compliance.zip -d ${HOME_DIR}/ \
+ && cp ${HOME_DIR}/oss_compliance/test/testOSSCompliance /usr/local/bin/testOSSCompliance \
+ && chmod +x /usr/local/bin/testOSSCompliance \
+ && chmod +x ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh \
+ && ${HOME_DIR}/oss_compliance/generate_oss_compliance.sh ${HOME_DIR} ${PYTHON} \
+ && rm -rf ${HOME_DIR}/oss_compliance* \
+ && rm -rf /tmp/tmp*
+
+RUN curl -o /license.txt  https://aws-dlc-licenses.s3.amazonaws.com/pytorch-1.11/license.txt
```
Comment (tejaschumbalkar) [inline pytorch/training/docker/1.12/py3/sdk2.6.0/Dockerfile.neuron:210]: change the license file to 1.12

File: pytorch/training/docker/1.12/py3/sdk2.6.0/Dockerfile.neuron.os_scan_allowlist.json [hunk 2]
```diff
@@ -0,0 +1,766 @@
+{
```
Comment (tejaschumbalkar) [inline pytorch/training/docker/1.12/py3/sdk2.6.0/Dockerfile.neuron.os_scan_allowlist.json:1]: is this file copied from previous version or are the file contents updated with the latest scan?

File: test/dlc_tests/sanity/test_pre_release.py [hunk 3]
```diff
@@ -292,7 +292,10 @@ def test_framework_and_neuron_sdk_version(neuron):
         tested_framework = tested_framework[len("huggingface_"):]
 
     if tested_framework == "pytorch":
-        tested_framework = "torch_neuron"
+        if "pytorch-training-neuron" in image:
```
Comment (tejaschumbalkar) [inline test/dlc_tests/sanity/test_pre_release.py:295]: is this compatible with previous versions of PT eg. PT1.11?

File: test/sagemaker_tests/pytorch/training/resources/neuron/mlp/requirements.txt [hunk 4]
```diff
@@ -1 +0,0 @@
-torchvision==0.12.0
```
Comment (tejaschumbalkar) [inline test/sagemaker_tests/pytorch/training/resources/neuron/mlp/requirements.txt:1]: can we delete the file if the contents are empty?

File: test/test_utils/__init__.py [hunk 5]
```diff
@@ -71,7 +71,7 @@ def get_ami_id_ssm(region_name, parameter_path):
 NEURON_UBUNTU_18_BASE_DLAMI_US_WEST_2 = get_ami_id_boto3(region_name="us-west-2", ami_name_pattern="Deep Learning Base AMI (Ubuntu 18.04) Version ??.?")
 UL20_PT_NEURON_US_WEST_2 = get_ami_id_boto3(region_name="us-west-2", ami_name_pattern="Deep Learning AMI Neuron PyTorch 1.11.0 (Ubuntu 20.04) ????????")
 # Since NEURON TRN1 DLAMI is not released yet use a custom AMI
-NEURON_TRN1_AMI_US_WEST_2 = "ami-0a610a15fcc9e0242"
+NEURON_TRN1_AMI_US_WEST_2 = "ami-06a5a60d3801a57b7"
```
Comment (tejaschumbalkar) [inline test/test_utils/__init__.py:74]: can we use ami pattern to fetch the latest ami instead of hardcoding it? eg
`get_ami_id_boto3(region_name="us-west-2", ami_name_pattern="<>")`

**Comment by tejaschumbalkar** [comment]
Please update PR description

File: pytorch/training/docker/1.12/py3/sdk2.6.0/Dockerfile.neuron (hunk 1 above)
Comment (aws-vrnatham) [reply pytorch/training/docker/1.12/py3/sdk2.6.0/Dockerfile.neuron]: Done

File: test/dlc_tests/sanity/test_pre_release.py (hunk 3 above)
Comment (aws-vrnatham) [reply test/dlc_tests/sanity/test_pre_release.py:334]: Yes

File: test/sagemaker_tests/pytorch/training/resources/neuron/mlp/requirements.txt (hunk 4 above)
Comment (aws-vrnatham) [reply test/sagemaker_tests/pytorch/training/resources/neuron/mlp/requirements.txt:1]: Done

File: test/test_utils/__init__.py (hunk 5 above)
Comment (aws-vrnatham) [reply test/test_utils/__init__.py]: Done.

**Comment by aws-vrnatham** [comment]
> 
Done

**Review by tejaschumbalkar — APPROVED** [review]
